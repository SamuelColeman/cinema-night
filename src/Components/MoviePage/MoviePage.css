.movie_page {
	border: 1px solid black;
	width: 31%;
	text-align: center;
	background-color: #E1E1E1;
}