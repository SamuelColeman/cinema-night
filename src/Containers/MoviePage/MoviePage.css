.movie_page {
	border: 1px solid black;
	text-align: center;
}

.movie__page--image {
	width: 50%;
	margin: 0;
}

.movie__page--card {
	display: flex;
	flex-direction: column;
	margin: 0;
}

h2,
h3 {
	color: white;
  padding: 3% 25%;
  margin: 0px;
}

.movie__page--title {
	color: white;
  padding: 1% 25%;
  font-size: 6vh;
}