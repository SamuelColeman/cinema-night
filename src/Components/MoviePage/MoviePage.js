import React from 'react';
import './MoviePage.css';
import { Link } from 'react-router-dom';

const MoviePage = ({ title, id, backdrop_path, poster_path, overview, release_date, vote_average }) => {
	return (
		<section className='movie_page' id={id}>
			<h1 className='movie__page--title'>{title}</h1>
			<img className="movie__page--img" src={poster_path} alt={title} />
			<h2>Release: {release_date}</h2>
			<h2>Rating: {vote_average}</h2>
			<img className="movie__page--img" src={backdrop_path} alt={title} />
			<h3>{overview}</h3>
			<Link to='/'>
				<button>Back</button>
			</Link>
		</section>
	)
}

export default MoviePage;