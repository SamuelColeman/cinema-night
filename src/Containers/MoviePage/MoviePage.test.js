import React from 'react';
 import { shallow } from 'enzyme';
 import { MoviePage } from './MoviePage';


describe('MoviePage', () => {
    let wrapper;
    let mockMovie = [
        {
          id: 456,
          title: 'Harry Potter and the Sorceres Stone',
          poster_path: 'https://pottersworld.com',
          release_date: '10-02-10',
          popularity: 462.91,
          vote_avarage: 3775,
          overview: 'yada yada',
        },
    ];

    beforeEach(() => {
      wrapper = shallow( <MoviePage 
        mockMovie = {mockMovie}
      />)
    })

    it('should match snapshot with all data being passed in correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })
 })

  