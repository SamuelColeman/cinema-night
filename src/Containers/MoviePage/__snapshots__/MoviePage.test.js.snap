// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage should match snapshot with all data being passed in correctly 1`] = `
<section
  className="movie_page"
>
  <h1
    className="movie__page--title"
  />
  <img
    className="movie__page--img"
  />
  <h2>
    Release: 
  </h2>
  <h2>
    Rating: 
  </h2>
  <img
    className="movie__page--img"
  />
  <h3 />
  <Link
    to="/"
  >
    <button>
      Back
    </button>
  </Link>
</section>
`;
